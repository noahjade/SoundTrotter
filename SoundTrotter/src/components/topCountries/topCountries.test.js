import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import TopCountries from ".";

test("Correctly renders default top countries component", () => {
    let countries = new Map();
    countries.set("A", { name: "A", code: "A", count: 5, artists: [] });
    countries.set("B", { name: "B", code: "B", count: 3, artists: [] });
    countries.set("C", { name: "C", code: "C", count: 6, artists: [] });

    const defaultTopCountriesComponent = shallow(<TopCountries countries={countries} />);
    expect(defaultTopCountriesComponent).toMatchSnapshot();
});