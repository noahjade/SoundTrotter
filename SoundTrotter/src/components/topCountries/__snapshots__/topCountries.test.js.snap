// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correctly renders default top countries component 1`] = `
<div
  className="container"
>
  <div
    className="title section"
  >
    You have recently listened to songs from 
     countries.
  </div>
  <div
    className="bottomSection"
  >
    <div
      className="title section"
    >
      There are still 
      NaN
       countries to explore!
    </div>
    <div
      className="buttonArea section"
    >
      <button
        onClick={[Function]}
      >
        Fill The Gaps
      </button>
    </div>
  </div>
  <div
    className="title"
  >
    Your Most Listened To Countries:
  </div>
  <div
    className="list section"
  >
    <div
      className="country"
      key="0"
    >
      1
      . 
      C
       (
      6
       artist
      s
      )
    </div>
    <div
      className="country"
      key="1"
    >
      2
      . 
      A
       (
      5
       artist
      s
      )
    </div>
    <div
      className="country"
      key="2"
    >
      3
      . 
      B
       (
      3
       artist
      s
      )
    </div>
  </div>
</div>
`;
